src/inc/til.h

```diff
@@ -18,6 +18,111 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
+        // Notes:
+        // - This and all below conversions make use of std::enable_if and a default parameter to disambiguate themselves.
+        //   enable_if will result in an <error-type> if the constraint within it is not met, which will make this
+        //   template ill-formed. Because SFINAE, ill-formed templated members "disappear" instead of causing an error.
+        template<typename TOther>
+        constexpr color(const TOther& other, std::enable_if_t<std::is_integral_v<decltype(std::declval<TOther>().R)> && std::is_integral_v<decltype(std::declval<TOther>().A)>, int> /*sentinel*/ = 0) :
+            r{ static_cast<uint8_t>(other.R) },
+            g{ static_cast<uint8_t>(other.G) },
+            b{ static_cast<uint8_t>(other.B) },
+            a{ static_cast<uint8_t>(other.A) }
+        {
+        }
+
+        // Method Description:
+        // - Converting constructor for any other color structure type containing integral r, g, b, a (case sensitive.)
+        template<typename TOther>
+        constexpr color(const TOther& other, std::enable_if_t<std::is_integral_v<decltype(std::declval<TOther>().r)> && std::is_integral_v<decltype(std::declval<TOther>().a)>, int> /*sentinel*/ = 0) :
+            r{ static_cast<uint8_t>(other.r) },
+            g{ static_cast<uint8_t>(other.g) },
+            b{ static_cast<uint8_t>(other.b) },
+            a{ static_cast<uint8_t>(other.a) }
+        {
+        }
+
+        // Method Description:
+        // - Converting constructor for any other color structure type containing floating-point R, G, B, A (case sensitive.)
+        template<typename TOther>
+        constexpr color(const TOther& other, std::enable_if_t<std::is_floating_point_v<decltype(std::declval<TOther>().R)> && std::is_floating_point_v<decltype(std::declval<TOther>().A)>, float> /*sentinel*/ = 1.0f) :
+            r{ static_cast<uint8_t>(other.R * 255.0f) },
+            g{ static_cast<uint8_t>(other.G * 255.0f) },
+            b{ static_cast<uint8_t>(other.B * 255.0f) },
+            a{ static_cast<uint8_t>(other.A * 255.0f) }
+        {
+        }
+
+        // Method Description:
+        // - Converting constructor for any other color structure type containing floating-point r, g, b, a (case sensitive.)
+        template<typename TOther>
+        constexpr color(const TOther& other, std::enable_if_t<std::is_floating_point_v<decltype(std::declval<TOther>().r)> && std::is_floating_point_v<decltype(std::declval<TOther>().a)>, float> /*sentinel*/ = 1.0f) :
+            r{ static_cast<uint8_t>(other.r * 255.0f) },
+            g{ static_cast<uint8_t>(other.g * 255.0f) },
+            b{ static_cast<uint8_t>(other.b * 255.0f) },
+            a{ static_cast<uint8_t>(other.a * 255.0f) }
+        {
+        }
+
+#ifdef D3DCOLORVALUE_DEFINED
+        constexpr operator D3DCOLORVALUE() const
+        {
+            return D3DCOLORVALUE{ r / 255.0f, g / 255.0f, b / 255.0f, a / 255.0f };

```

Comment:
do we need to cast `r`/`g`/`b`/`a` to floats before dividing in C++?

Reply:
it does not appear so

Reply:
k